instrumentation/spring/spring-webflux-5.0/javaagent/src/test/java/SpringWebfluxTest.java

```diff
@@ -0,0 +1,841 @@
+import static io.opentelemetry.semconv.trace.attributes.SemanticAttributes.NET_TRANSPORT;
+import static io.opentelemetry.semconv.trace.attributes.SemanticAttributes.NetTransportValues.IP_TCP;
+import static org.junit.jupiter.api.Named.named;
+
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.instrumentation.testing.junit.AgentInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.sdk.testing.assertj.TraceAssert;
+import io.opentelemetry.sdk.trace.data.StatusData;
+import io.opentelemetry.testing.internal.armeria.client.WebClient;
+import io.opentelemetry.testing.internal.armeria.common.AggregatedHttpResponse;
+import io.opentelemetry.testing.internal.armeria.common.HttpStatus;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.boot.web.embedded.netty.NettyReactiveWebServerFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import server.EchoHandlerFunction;
+import server.FooModel;
+import server.SpringWebFluxTestApplication;
+import server.TestController;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+    classes = {
+      SpringWebFluxTestApplication.class,
+      SpringWebfluxTest.ForceNettyAutoConfiguration.class
+    })
+public class SpringWebfluxTest {
+  @TestConfiguration
+  static class ForceNettyAutoConfiguration {
+    @Bean
+    NettyReactiveWebServerFactory nettyFactory() {
+      return new NettyReactiveWebServerFactory();
+    }
+  }
+
+  @RegisterExtension
+  static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  private static final String INNER_HANDLER_FUNCTION_CLASS_TAG_PREFIX =
+      SpringWebFluxTestApplication.class.getName() + "$";
+  private static final String SPRING_APP_CLASS_ANON_NESTED_CLASS_PREFIX =
+      SpringWebFluxTestApplication.class.getSimpleName() + "$";
+
+  // can't use @LocalServerPort annotation since it moved packages between Spring Boot 2 and 3
+  @Value("${local.server.port}")
+  private int port;
+
+  private WebClient client;
+
+  @BeforeEach
+  public void beforeEach() {
```

Comment:
JUnit 5 test classes & methods don't have to be public
```suggestion
  void beforeEach() {
```